DOCS-1364: Document creating datasets (#2239)
viamrobotics/docs: docs/manage/data/dataset.md

---
title: "Label data and create datasets"
linkTitle: "Create Datasets"
description: "Label data and create datasets for managing data and creating machine learning models."
weight: 39
type: "docs"
tags: ["data management", "cloud", "sync"]
images: ["/manage/data/label-dog.gif"]
aliases:
  - "/manage/data/label"

> # SME: Tahiya Salam and Alexis Wei

Reviewer: Can make this Tahiya + me

> # SME: Tahiya Salam and Alexis Wei

Author: ```suggestion
# SME: Tahiya Salam and Alexa Greenberg
```

To delete a tag, click the cross next to the corresponding label.

## Datasets

{{< alert title="Support" color="info" >}}
Datasets currently only work for images.
{{< /alert >}}

### Create a dataset

To create a dataset, go to the [**DATA** tab](https://app.viam.com/data/view) in the Viam app.

> On the **DATASETS** subtab, click **Create dataset**, enter a name for your dataset and click **Create dataset** again.

Reviewer: We can add it in explicitly when https://viam.atlassian.net/browse/DATA-2166 is resolved, but might be worth mentioning that it can be created directly on the image, the same way that you add data to a dataset below.

Though I know it's already written below that `You can select an existing dataset or create a new one from this dropdown` so perhaps too redundant?

> On the **DATASETS** subtab, click **Create dataset**, enter a name for your dataset and click **Create dataset** again.

Author: ```suggestion
On the [**DATASETS** subtab](https://app.viam.com/data/datasets), click **Create dataset**, enter a name for your dataset and click **Create dataset** again.
```

![Create dataset UI](/manage/data/create-dataset.png)

> you can filter available images, using the **Filtering** menu and select the attributes that match where, how, and when the data was collected.

Author: ```suggestion
```

### Add or remove data from a dataset

To add or remove data from a dataset, go to the [**DATA** tab](https://app.viam.com/data/view) in the Viam app.

On the **Images** subtab, you can filter available images, using the **Filtering** menu and select the attributes that match where, how, and when the data was collected.

> To add image to a dataset, click on the image and select the dataset in the Datasets dropdown.

Reviewer: [supernit] an image

> To add image to a dataset, click on the image and select the dataset in the Datasets dropdown.

Author: ```suggestion
To add an image to a dataset, click on the image and select the dataset in the Datasets dropdown.
```

You can select an existing dataset or create a new one from this dropdown.

{{<gif webm_src="/manage/data/add-to-dataset.webm" mp4_src="/manage/data/add-to-dataset.mp4" alt="Add image to dataset">}}

To remove an image from a dataset click on the delete button next to the dataset.

{{<gif webm_src="/manage/data/delete-from-dataset.webm" mp4_src="/manage/data/delete-from-dataset.mp4" alt="Remove from dataset">}}

### View the data in a dataset

> Once you have added data to your dataset, you can view the data from the **DATASETS** tab by clicking on your dataset.

Reviewer: You can also view the data by clicking on the dataset directly from the image sidebar, since each listed dataset on the image is a direct link into that dataset page.

and [nit], perhaps 'by clicking on your dataset name'?

> Once you have added data to your dataset, you can view the data from the **DATASETS** tab by clicking on your dataset.

Author: ```suggestion
Once you have added data to your dataset, you can view the data from the [**DATASETS** subtab](https://app.viam.com/data/datasets) by clicking on your dataset.
```

> You can also add or remove labels or tags from this menu or remove data from the dataset by clicking on an image and using the appearing menu.

Reviewer: This took me a few reads — perhaps:

`
You can click on any image in the dataset and use its appearing menu in order to add or remove bounding box labels or tags, or remove that image from the dataset
`

Author: sounds good thx!

> You can also add or remove labels or tags from this menu or remove data from the dataset by clicking on an image and using the appearing menu.

Author: ```suggestion
You can click on any image in the dataset and use its appearing menu in order to add or remove bounding box labels or tags, or remove that image from the dataset.
```

### Train a model on a dataset

To train a model on a dataset, see [Train a model](/manage/ml/train-model/).

### Delete a dataset

> To delete a dataset, right click on the dataset and click on the **Delete** button that appears.

Reviewer: To make it clear that this is on the listed dataset page, perhaps 'right click on the dataset in https://app.viam.com/data/datasets'?

> To delete a dataset, right click on the dataset and click on the **Delete** button that appears.

Author: ```suggestion
To delete a dataset, go to the [**DATASETS** subtab](https://app.viam.com/data/datasets), right click on the dataset and click on the **Delete** button that appears.
```
